daemon/networkdriver/bridge/driver.go

```diff
@@ -140,6 +194,16 @@ func InitDriver(job *engine.Job) engine.Status {
 		if err := ioutil.WriteFile("/proc/sys/net/ipv4/ip_forward", []byte{'1', '\n'}, 0644); err != nil {
 			job.Logf("WARNING: unable to enable IPv4 forwarding: %s\n", err)
 		}
+
+		if enableIPv6 {
+			// Enable IPv6 forwarding
+			if err := ioutil.WriteFile("/proc/sys/net/ipv6/conf/default/forwarding", []byte{'1', '\n'}, 0644); err != nil {
+				job.Logf("WARNING: unable to enable IPv6 default forwarding: %s\n", err)
+			}
+			if err := ioutil.WriteFile("/proc/sys/net/ipv6/conf/all/forwarding", []byte{'1', '\n'}, 0644); err != nil {
```

Comment:
To describe the current behavior:
Router advertisement sets up a /128 IPv6 address for the host - let's say on eth0.
If you want to use Docker with IPv6 your network router should be set up to route a /64 to your host's eth0. Then you can assign an IPv6 address from this range manually to your eth0. Router advertisement does not work for this scenario.
If you have two ethernet devices, eth0 with a /64 subnet and eth1 with an /128 address assigned via RA, yes, this would be a problem. Docker would prevent eth1 to get an address via RA when started with the default `-ip-forward=true`.
If you have just one eth0 and want to use RA and a /128 address you cannot route IPv6 traffic to your virtual docker network. So you would have to start Docker with `-ip-forward=false`.
There is no IPv6 NAT.